src/GenesisMakeFileWindow.cpp

```diff
@@ -137,21 +158,67 @@ void GenesisMakeDirWindow::MessageReceived(BMessage* message)
 }
 
 ////////////////////////////////////////////////////////////////////////
-bool GenesisMakeDirWindow::CreateFolder(const char *dirpath, const char *dirname)
+bool GenesisMakeFileWindow::CreateFolder(const char *dirpath, const char *dirname)
 ////////////////////////////////////////////////////////////////////////
 {
 	BString dir;
 	dir.SetTo(dirpath);
 	dir+="/";
 	dir+=dirname;
 
-	if (create_directory(dir.String(), 0777)==B_OK)		// TODO: jo a 0777?
+	BEntry *entry = new BEntry(dir);
+	if (entry->Exists() && entry->IsDirectory())
+	{
+		BAlert *alert = new BAlert("Cannot create new folder","Folder already exists","OK",NULL,NULL,B_WIDTH_AS_USUAL,B_INFO_ALERT);
```

Comment:
The first argument to BAlert is currently not exposed in the GUI, but might one day (by some Decorator). I'd go for a short title, like "_Error creating folder_".

The text of the alert itself can be more user-centric, elaborate. Like: "_A folder with that name already exists._"
Maybe even but the folder name into the text: "_The folder 'name-of-folder' already exists._"